src/main/java/org/broadinstitute/consent/http/resources/DatasetResource.java

```diff
@@ -280,6 +281,30 @@ public Response getDataset(@PathParam("datasetId") Integer datasetId){
         }
     }
 
+    @GET
+    @Path("/batch")
+    @Produces("application/json")
+    @PermitAll
+    public Response getDatasets(@QueryParam("ids") List<Integer> datasetIds){
+        try {
+            List<Dataset> datasets = datasetService.getDatasets(datasetIds);
+
+            List<Integer> foundIds = datasets.stream().map(Dataset::getDataSetId).toList();
+            if (!foundIds.containsAll(datasetIds)) {
```

Comment:
So, I may have steered you poorly - I didn't actually test the code I used as an example. Turns out `List. containsAll()` [has poor performance](https://stackoverflow.com/questions/10199772/what-is-the-cost-of-containsall-in-java). The suggestion is to use a hash instead,
```
if (!new HashSet<>(foundIds).containsAll(datasetIds)) {
```